src/main/scala/mesosphere/marathon/storage/migration/Helper.scala

```diff
@@ -33,3 +46,63 @@ trait MaybeStore {
   def maybeStore(persistenceStore: PersistenceStore[_, _, _]): Option[ZkPersistenceStore] = findZkStore(persistenceStore)
 
 }
+
+/**
+  * Provides a common pattern to migrate instances. This makes testing migrations easier.
+  */
+object InstanceMigration extends MaybeStore with StrictLogging {
+
+  /**
+    * This function traverses all instances in ZK and passes them through the migration flow before saving the updates.
+    */
+  def migrateInstances(
+    instanceRepository: InstanceRepository,
+    persistenceStore: PersistenceStore[_, _, _],
+    migrationFlow: Flow[JsValue, Instance, NotUsed])(implicit ctx: ExecutionContext, mat: Materializer): Future[Done] = {
+
+    val countingSink: Sink[Done, NotUsed] = Sink.fold[Int, Done](0) { case (count, Done) => count + 1 }
+      .mapMaterializedValue { f =>
+        f.map(i => logger.info(s"$i instances migrated"))
+        NotUsed
+      }
+
+    maybeStore(persistenceStore).map { store =>
+      instanceRepository
+        .ids()
+        .mapAsync(1) { instanceId =>
+          store.get[Id, JsValue](instanceId)
```

Comment:
To improve debuggability:
```suggestion
          store
          .get[Id, JsValue](instanceId)
          .map { 
            case Some(jsValue) => Some(jsValue)
            case None => logger.error(s"Failed to load an instance for $instanceId. It will be ignored"; None
          }
```